src/System.Net.Http/tests/FunctionalTests/ResponseStreamTest.cs

```diff
@@ -68,6 +69,26 @@ public ResponseStreamTest(ITestOutputHelper output)
             }
         }
 
+        [Fact]
+        public async Task ReadAsStreamAsync_Cancel_TaskIsCanceled()
+        {
+            var cts = new CancellationTokenSource();
+
+            using (var client = new HttpClient())
+            {
+                HttpResponseMessage response =
+                    await client.GetAsync(HttpTestServers.RemoteEchoServer, HttpCompletionOption.ResponseHeadersRead);
+                Stream stream = await response.Content.ReadAsStreamAsync();
+
+                var buffer = new byte[2048];
+                Task task = stream.ReadAsync(buffer, 0, buffer.Length, cts.Token);
```

Comment:
>then simply verifying that it completes successfully or due to cancellation sounds good.

Is there a version of `Task.Wait()` that doesn't throw?  I basically want to wait for the task to move to a terminal state. But I would rather it not throw and then I can check the Task statuses directly.

>Do we also need to dispose of the stream and/or response?

For completeness, I'll dispose of those similar to the newer patterns we're using in the tests.